src/System.Text.Primitives/System/Text/InvariantParser_byte.cs

```diff
@@ -35,19 +35,23 @@ public static bool TryParse(byte[] utf8Text, int index, out byte value, out int
                         return true; // otherwise return true
                     }
                 }
-                byte candidate = (byte)(value * 10); // left shift the value
-                candidate += (byte)(nextByte - '0'); // parse the current digit to a byte and add it to the temporary value
-                if (candidate >= value) // if it was a digit 0-9, this should be true
+                try
                 {
-                    value = candidate;
+                    byte candidate = (byte)(value * 10); // left shift the value
+                    candidate = checked((byte)(candidate + nextByte - '0')); // parse the current digit to a byte and add it to the temporary value
```

Comment:
True. I'm is there any real reason we have it split into two lines? I'll just collapse it if not.